ext/opcache/zend_shared_alloc.c

```diff
@@ -336,6 +336,8 @@ static size_t zend_shared_alloc_get_largest_free_block(void)
 
 void *zend_shared_alloc(size_t size)
 {
+	ZEND_ASSERT(ZCG(locked));
```

Comment:
I think this assertion makes sense, although a few lines down that condition is checked, and fatal error is triggered if there is no log. Maybe we should remove the preprocessor guard ('#if 1') for that, though.

Reply:
Oh yes, this `#if 1` is confusing - though I'd prefer to remove this section and use only `ZEND_ASSERT`, because this is not a fatal runtime error, but a bug, and that's what assertions are for.

Reply:
> because this is not a fatal runtime error, but a bug, and that's what assertions are for.

Ah, indeed!